Always check configuration before gathering data.

wenottingham opened the pull request:
We shouldn't perform expensive operations if we won't be able to send it.

wenottingham commented:
Works in brief testing, starting more detailed regression testing.

kdelee commented on awx/main/analytics/core.py:
```diff
@@ -26,6 +26,14 @@
+        if not getattr(settings, key, None):
+            logger.error('%s is not set' % (key,))
+            return False
+    return True
```
@wenottingham this covers case where username/password not set but does not cover case where they are invalid. Do we want to try that out?

Also, if I'm reading this correctly we would still allow the user to enabled automation analytics w/o username/password or invalid username/password as we do today but now we would log this error when we then are unable to proceed because of missing auth.

I'm guessing we would want to make a backlog item for making this more visible in the UI if automation analytics is enabled but failing because of this?

wenottingham replied:
Testing validity would be making a random connection that didn't send anything, I'm not sure yet if i want to cause logs/errors on the insights data gathering side for just testing a connection.

Correct, we may want to surface this higher in the future; this is just to stop it from even trying and wasting resources if they didn't configure it right.

wenottingham commented:
This ain't it, rewriting.

wenottingham committed:
Always check configuration before gathering data.

We shouldn't perform expensive operations if we won't be able to send it.
Only log at debug level, otherwise every node will log this every 5
minutes.
